Archive/Draft/RFCxxxx-PS7-Bootstrapper.md

```diff
@@ -0,0 +1,256 @@
+version is available, but requires elevation or an administrator to install.
+
+>[Note] Need to verify that Squirrel supports admin installs for system-wide access
+
+### Windows Terminal
+
+Windows Terminal is now a default installed app and has code to detect PS7.
+Currently, it will not detect the prescence of `pwsh.exe` bootstrapper if it exists in system32.
+We may submit a PR to update their detection logic so that the bootstrapper is available as the default shell.
+
+### Offline scenarios
+
+Need to look into integration with [Windows ADK (Assessment and Deployment Kit)](https://docs.microsoft.com/en-us/windows-hardware/get-started/)
+to support pre-provisioning Windows images that already include PS7 rather than bootstrapping at runtime where the machine may be on a factory
+floor (WinPE) without internet access.
+
+### Remoting endpoint
+
+By default, neither WinRM nor SSHD will be setup for PS7 remoting.
+So when remoting from one Windows machine into another, it would not connect to PS7 unless the user has manually configured it
+(and thus also installed PS7).
+
+We may consider adding capability to make it easy to setup PS7 remoting for large deployments in the future.
+For example, maybe `pwsh.exe --enable-ssh-endpoint` and `pwsh.exe --enable-winrm-endpoint` as special switches only for the bootstrapper.
+
+## Alternate Proposals and Considerations
+
+### Reducing size
+
+The bootstrapper itself is small, but installation would still carry a full PS7 package.
+Details of reducing the size of PS7 post-install is outside the scope of this specific RFC.
+
+### Auto-update when used non-interactively
+
+Current design is based on auto-update (unless configuration is set not to) even if the usage is via automation.
+For example, remotely or using task scheduler to execute a script.
+A design decision could be made to not auto-update if we know the usage is not interactive,
+but current design is to keep PS7 updated even in those cases unless configured otherwise.
+
+### Feature-on-Demand
+
+A Windows Feature-on-Demand (FOD) is similar to what is being proposed in that you would install the feature from the internet.
+However, a FOD needs to be explicitly installed to be used and is also locked to a specific version built for that version of Windows.
+In this case, it's critical to allow use of `pwsh.exe` right out of the box and have it execute a customer script and also
+because PS7 aligns with .NET lifecycle, install the latest stable version.
+
+### Windows Store MSIX pkg
+
+Some applications have a shortcut that is shipped inbox and when started installs from the Windows Store.
+This presents two problems:
+
+- This works fine for interactive use, but for automation, they now have to bootstrap PS7 before they can use PS7 to bootstrap other configuration
+- Windows Store is not supported on Windows Server which is required
```

Comment:
You could alternatively investigate using [AppInstaller](https://docs.microsoft.com/en-us/windows/msix/app-installer/app-installer-file-overview). Or at least evaluate it and include the "why not" in this section.

Reply:
Isn't that MSIX specific?  MSIX is not supported on Windows Server

Reply:
@SteveL-MSFT MSIX is supported on Windows Server. You might be thinking of the AppInstaller app (which is separate from the .appinstaller manifest).